src/input.ts

```diff
@@ -3,21 +3,29 @@ import fs from "fs";
 import path from "path";
 import type { Collection } from "./types";
 
-interface YamlInput {
+export interface NetlifyCMSConfig {
   collections?: Collection[];
 }
 
-export const loadConfig = (filePath: string): Collection[] => {
-  if (!path.extname(filePath).match(/\.ya?ml$/)) {
-    throw new Error("Invalid filetype, must be an yaml file");
+export const loadConfig = (config: string | NetlifyCMSConfig): Collection[] => {
+  let data;
+
+  if (typeof config === "object") {
+    data = config;
   }
 
-  const file = fs.readFileSync(path.resolve(process.cwd(), filePath), "utf8");
+  if (typeof config === "string") {
```

Comment:
The loading of a config file could be extracted into its own function, e.g. `loadConfigFile(filePath: string): NetlifyCMSConfig` for better readability.